admin/src/js/controllers/edit-user.js

```diff
@@ -298,12 +298,22 @@ angular.module('controllers').controller('EditUserCtrl',
                   $uibModalInstance.close();
                 })
                 .catch(function(err) {
-                  $scope.setError(err, 'Error updating user');
+                  if (err.status === 400) {
+                    Translate('username.taken', { username: $scope.editUserModel.username }).then(function(value) {
```

Comment:
I think more errors with status 400 could be received from API. 
Also, if it's easier to implement, you can change the error message not not include the actual username.